inc/class-tmsc.php

```diff
@@ -1,106 +1,314 @@
 <?php
-if ( ! class_exists( 'TMSC' ) ) {
 
-	class TMSC {
+namespace TMSC;
 
-		private static $instance;
+class TMSC {
 
-		// The default zones we load objects into
-		public $default_zones = array();
+	private static $instance;
 
+	// The default zones we load objects into
+	public $default_zones = array();
+
+	public $processors = array();
+
+	/**
+	 * Constructor
+	 *
+	 * @params string $name
+	 * @params url $name optional
+	 * @return void
+	 */
+	public function __construct() {
+		/* Don't do anything, needs to be initialized via instance() method */
+	}
+
+	/**
+	 * Instatiate our instance.
+	 */
+	public static function instance() {
+		if ( ! isset( self::$instance ) ) {
+			self::$instance = new \TMSC\TMSC();
```

Comment:
Same as above, you shoudl already be in the namespace so I think you can just do `TMSC()`.